lib/srv/desktop/tdp/proto.go

```diff
@@ -1418,6 +1471,17 @@ func writeUint64(b *bytes.Buffer, v uint64) {
 	b.WriteByte(byte(v))
 }
 
+// tdpMaxNotificationMessageLength is somewhat arbitrary, as it is only sent *to*
+// the browser (Teleport never receives this message, so won't be decoding it)
+const tdpMaxNotificationMessageLength = 10240
+
+// tdpMaxPathLength is currently just an arbitrary value, more research is required
+// (see https://github.com/gravitational/teleport/issues/14950)
+const tdpMaxPathLength = 10240
+
+const maxClipboardDataLength = 1024 * 1024    // 1MB
+const tdpMaxFileReadWriteLength = 1024 * 1024 // 1MB
```

Comment:
These aren't exactly the same limits. The limit in the Rust code is for a message received on a vchannel -- it's 2MB because that let's us receive 1MB of raw read/write data with plenty of wiggle room for the additional message fields and headers. This limit is for just the raw read/write data.